posts/2021-02-19-mpcontextpropagation-requesttiming-21003-beta.adoc

```diff
@@ -0,0 +1,314 @@
+
+When a hung request is detected in the Request Timing feature, a warning message is written in the messages log file along with a dump of the tree of events that happened during the request. Following that, a set of three thread dumps will be initiated, 1 minute apart. After the completion of the three thread dumps, further set of three thread dumps are created only if new requests are detected to be hanging.
+
+Some operations teams do not want thread dumps to be generated, whenever a hung request is detected in the requestTiming-1.0 feature, because there will be unwanted java thread dumps for requests that are known to be normally long and the process of generating them will impact performance. In previous Open Liberty releases, there was no option to disable the thread dumps from being generated each time a hung request is detected.
+
+You can now control whether the Request Timing feature collects thread dumps dumps or not when a hung request is detected. By simply configuring the **NEW** `enableThreadDumps` Request Timing server configuration attribute, to false, thread dumps will not be created during hung requests. Thread dumps will be created, by default, if the new server configuration attribute is set to true or not specified at all.
+   
+The new Request Timing server configuration attribute can be configured in your server.xml as follows:
+
+[source, xml]
+----
+<requestTiming includeContextInfo="true" slowRequestThreshold="120s" hungRequestThreshold="10s" sampleRate="1" enableThreadDumps="false"></requestTiming>`
+----
+
+
+The `enableThreadDumps` server configuration attribute can also be used in embedded Request Timing sub-elements: 
+`<servletTiming/>` or `<jdbcTiming/>`, as follows:
+
+[source, xml]
+----
+<requestTiming includeContextInfo="true" slowRequestThreshold="120s" hungRequestThreshold="10s" sampleRate="1">
+    <servletTiming appName="MyApp" servletName="MyServletApp" slowRequestThreshold="100s" hungRequestThreshold="5s" enableThreadDumps="false"/>
+</requestTiming>`
+----
+
+NOTE: An embedded `<servletTiming/>` or `<jdbcTiming/>` configuration in the server.xml file overrides the configured slow and hung request threshold that are defined in `<requestTiming/>`. 
+
+For more information on the Request Timing feature, please refer to the following documentations:
+- link:{url-prefix}/docs/21.0.0.1/reference/feature/requestTiming-1.0.html[Open Liberty Documentation on requestTiming-1.0]
+- {url-prefix}/docs/21.0.0.1/reference/config/requestTiming.html
```

Comment:
remove me

Reply:
Done, not removed but made into an actual link